### ISSUE TITLE
Allow Messages and Phone apps to have different blocked numbers

### ISSUE THREAD

**Issue by antodc** [FossifyOrg/Messages#237]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

I receive several spam/ads call every day, but I received almost no spam/ads messages in my life. So, I have a long list of blocked numbers and rules for phone calls that I would like not to apply to messages.

### Why do you want this feature?

It could happen that one of the rules I set for the phone call blocks by mistake also the number of a friend and I would like him/her to be able to contact me by messages if necessary.
More in general, this feature would increase the flexibility in the usage of two different apps.

**Comment by Cwpute** [comment]
Maybe this is not so much a matter of having two lists of blocked contacts, but rather let users avoid blocking their contacts ?

For example: blocking a number that is in your contact-list would trigger a warning saying something like `[Contact Name] is in your contacts list. Block this number anyway ?`.
Would that solution solve your problem ?

**Comment by antodc** [comment]
I think that it should be a basic feature of blocked numbers. If I am not mistaken, I have read some feature/bug request with a similar suggestion.
In my case it would not help. I would like to keep the phone calls I receive as restricted as possible while still being able to receive messages from (almost) everyone.

**Comment by Cwpute** [comment]
How does it happen when you block a number by mistake ?
Is there a missing information or prompt somewhere in the process that could help you avoid these mistakes ?

**Comment by antodc** [comment]
Let me provide an example.
I have blocked the number(s) +393471821* because I receive several spam calls from numbers like that. However, it may happen some day that someone important with a number like that wants to contact me (say someone from my kid’s school that has never called me before). If also messages are blocked by that rule there is no way he can contact me.

**Comment by Cwpute** [comment]
I see, so this is not a mistake you're making, it's actually your catch-all blocking rule that's a little bit too powerful.

The reason i'm trying to understand the circumstances around your feature request is that this might require a lot of work for a pretty edge case, and separate blacklist for each app will be hard to explain and present to regular users who only seldom use the black list functionality anyway.
I don't know about the feasibility of this next suggestion, but let's hear it out anyway: what about options on the way the blacklist works ? Users could choose whether ALL blacklisted numbers would be able to call them, send them sms messages, leave vocal messages, or a combination of these options.

Finally, have you tried other external solutions ?
If you are in the US or Canada, have you tried using the Carrion app ? https://gitlab.com/divested-mobile/carrion
If you are in France, have you tried registering your number to Bloctel ? https://www.bloctel.gouv.fr/
Look up your country's legal practice, you might find something to help you as several other countries provide protection against unsollicited calls like these.

**Comment by Aga-C** [comment]
> I don't know about the feasibility of this next suggestion, but let's hear it out anyway: what about options on the way the blacklist works ? Users could choose whether ALL blacklisted numbers would be able to call them, send them sms messages, leave vocal messages, or a combination of these options. Would that solve your problem ?

The current block list is a system one, so it will always block calls. We could add an option whether it blocks SMS or not, but then we would need to keep a separate list for blocking SMS. And that's exactly the feature request here.

**Comment by antodc** [comment]
@Cwpute, @Aga-C. Thank you both for your information/suggestions. Instead of creating two separate lists, that I understand may be messy, what about adding a switch close to each blocking rule to enable/disable it for the specific app? Or, maybe, is this internally equivalent to creating two lists anyway? I do not know.
In any case, I do not what to push for any eccessive extra effort from your side. You are already doing an amazing work with this suite of apps and I am aware that this would be a somewhat minor improvement, if any.

**Comment by Aga-C** [comment]
> Or, maybe, is this internally equivalent to creating two lists anyway?

Exactly. We would need to keep two lists internally, so it doesn't matter if user will see two lists or just have switches in the current one.

**Comment by Aga-C** [comment]
@rmj-maker When implementing this feature, the option to block all unknown numbers should also be separated.
